griptape/tools/google_sheets/tool.py

```diff
@@ -0,0 +1,400 @@
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error listing spreadsheets: {e}")
+
+        return ListArtifact(spreadsheets)
+
+    @activity(
+        config={
+            "description": "Can be used to creates a new Google Sheet with the specified title.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "title", description="The title of the new spreadsheet"
+                    ): str
+                }
+            ),
+        }
+    )
+    def create_spreadsheet(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        from googleapiclient.errors import HttpError
+        from google.auth.exceptions import MalformedError
+
+        title = params["title"]
+
+        try:
+            service = self._build_client(
+                scopes=self.SHEETS_SCOPES,
+                service_name="sheets",
+                version="v4",
+                owner_email=self.owner_email,
+            )
+
+            spreadsheet_body = {"properties": {"title": title}}
+
+            spreadsheet = (
+                service.spreadsheets().create(body=spreadsheet_body).execute()
+            )
+            return InfoArtifact(
+                f"Spreadsheet created with ID: {spreadsheet['spreadsheetId']}"
+            )
+
+        except HttpError as e:
+            logging.error(e)
+            return ErrorArtifact(
+                f"error creating spreadsheet due to http error: {e}"
+            )
+        except MalformedError as e:
+            logging.error(e)
+            return ErrorArtifact(
+                f"error creating spreadsheet due to malformed credentials: {e}"
+            )
+
+    @activity(
+        config={
+            "description": "Can be used to downloads multiple spreadsheets based on provided file paths",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_paths",
+                        description="List of file paths to the spreadsheets.",
+                    ): list[str],
+                    Literal(
+                        "mime_type",
+                        description="The MIME type for the file format to export to.",
+                    ): str,
+                }
+            ),
+        }
+    )
+    def download_spreadsheets(self, params: dict) -> ListArtifact:
+        from google.auth.exceptions import MalformedError
+        from googleapiclient.errors import HttpError
+
+        file_paths = params["file_paths"]
+        mime_type = params["mime_type"]
+
+        export_mime_mapping = {
+            "text/csv": "text/csv",
+            "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        }
+
+        downloaded_files = []
+
+        if mime_type not in export_mime_mapping:
+            logging.error("Unsupported mime type for downloading")
```

Comment:
This should probably return an `ErrorArtifact`.

Reply:
Ok!